### DISCUSSION TITLE
fastify not rendering an API reference for a given OpenAPI/Swagger spec URL

### DISCUSSION THREAD

**Discussion by cristianabs** [scalar/scalar#1702] [Q&A] [answered]
Hello,

I have followed all the steps from the following link: https://github.com/scalar/scalar/blob/main/documentation/fastify.md

It works to get the information automatically from fastify.swagger, but if I want to use an external API reference it does not load anything in the browser.

I have added the following configuration:

fastify.register(require('@scalar/fastify-api-reference'), {
  routePrefix: '/reference',
  configuration: {
    title: 'Our API Reference',
    spec: {
      url: '/openapi.json',
    },
  },
})

My openapi.json file is in the same directory as my index.js.

Also, I have removed the code related to fastify.register(FastifySwagger ..) and fastify.put( .. ) and nothing, there is still no information exposed, although any other configurations work aside from "isEditable: true" which does not allow me to make any changes.

<img width="870" alt="Screenshot 2024-05-11 at 10 46 39" src="https://github.com/scalar/scalar/assets/6587063/e2e89306-3d49-47a1-bca6-b240c816f904">
<img width="1630" alt="Screenshot 2024-05-11 at 10 46 44" src="https://github.com/scalar/scalar/assets/6587063/4917eb4b-f303-4b10-b3db-3a0b76afa7fb">

Finally, I also tried using spec: { content: '{ … }' }, but any content json or yaml is not accepted and there is no specification regarding how it should be written.

So I would appreciate any clue from someone to help me. I am very new to justify and I am learning about this topic.

Thank you.

**Comment by marclave** [comment]
thanks for reporting this @cristianabs !

is there anything in your console, are you using helmet? ✨ just trying to see if theres an error message in the console to debug cause i could get it running locally 🤔

**Comment by cristianabs** [comment]
Hi @marclave,

I am not using helmet and my console is empty, my terminal is showing the following in the image:

<img width="1710" alt="Screenshot 2024-05-11 at 15 17 41" src="https://github.com/scalar/scalar/assets/6587063/a3f64c45-c240-4265-a96d-ff880d3833b2">

And I don't have anything else, I am literally following the steps written in this document: https://github.com/scalar/scalar/blob/main/documentation/fastify.md

Thanks

**Reply by marclave** [reply]
hey @cristianabs https://github.com/marclave/fastify-scalar-repro

could you clone this repo and run the following:

```
npm install
npm run dev
```

and see if that works?

**Comment by cristianabs** [comment]
hey @marclave  yes it works the same as in the documentation, the issue I find is when I am trying to use an external openapi.json file and not the information in the code under fastify.put, because it is the only information present in the web browser. 

I mean this part of the code:

await fastify.register(ScalarApiReference, {
  routePrefix: '/reference',
  configuration: {
    title: 'Our API Reference',
    spec: {
    url: '/openapi.json',
    },
    },
})

Is not fetching the data from the URL (local file I reckon)

**Reply by marclave** [reply]
ah now i understand! sorry, you would need to serve the local openapi file!

you would need to serve the file locally like so https://github.com/fastify/fastify-static then you can reference an external file :)

**Reply by cristianabs** [reply, marked as answer]
@marclave sorry for the confusion, perfect now it is working!! thank you :)
